jupyterhub/orm.py

```diff
@@ -64,6 +64,8 @@ class Server(Base):
     """
     __tablename__ = 'servers'
     id = Column(Integer, primary_key=True)
+    
+    name = Column(Unicode(32), unique=True) # this is a unique reference to the server, probably a UUID4
```

Comment:
So I think the main problem is to decide what we want to use as server/{name}

UUID, primary key (id), or something else ?

About the constraint you are right, but we don't have the name of the user inside the Server table so I think we cannot put like Unique(user_name, server_name).
But as you said we can mix both name together and put that field like  unique.